portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -861,18 +861,16 @@ public Group deleteGroup(Group group) throws PortalException {
 					group.getGroupId(), RoleConstants.TYPE_SITE);
 			}
 			else {
-				if (!group.isStagingGroup() || group.isStagedRemotely()) {
```

Comment:
> Drew, did you find existing site role assignments? Do you know why?
@JorgeFerrer yes that is actually the cause of the issue.  There are UserGroupRoles for the staging site that never get removed when the group is deleted, which causes ModelListenerExceptions to be thrown when the security audit modules are present.

@topolik @arbetts would it make more sense in this case to prevent the UserGroupRoles from being added to the staging site in the first place?